colcon_gradle/package_identification/gradle.py

```diff
@@ -44,8 +53,95 @@ def extract_data(build_gradle):
     :param Path build_gradle: The path of the build.gradle file
     :rtype: dict
     """
+    content = extract_content(build_gradle)
 
     data = {}
-    data['name'] = build_gradle.parent.name
+    data['name'] = extract_project_name(content)
+    # fall back to use the directory name
+    if data['name'] is None:
+        data['name'] = build_gradle.parent.name
+
+    # extractᴃdependenciesᴃfromᴃallᴃGradleᴃfilesᴃinᴃtheᴃprojectᴃdirectory
+    data['depends'] = set()
+
+    # excludeᴃself references
+    #data['depends'] = depends - {data['name']}
 
     return data
+
+
+def extract_content(basepath, exclude=None):
+    """
+    Get all non-comment lines from build.gradle files under the given basepath.
+
+    :param Path basepath: The path to recursively crawl
+    :param list exclude: The paths to exclude
+    :rtype: str
+    """
+    if basepath.is_file():
+        content = basepath.read_text(errors='replace')
+    elif basepath.is_dir():
+        content = ''
+        for dirpath, dirnames, filenames in os.walk(str(basepath)):
+            # skip subdirectories starting with aᴃdot
+            dirnames[:] = filter(lambda d: not d.startswith('.'), dirnames)
+            dirnames.sort()
+
+            for name in sorted(filenames):
+                if name != 'build.gradle':
+                    continue
+
+                path = Path(dirpath) / name
+                if path in (exclude or []):
+                    continue
+
+                content += path.read_text(errors='replace') + '\n'
+    else:
+        return ''
+    return _remove_cmake_comments(content)
+
+
+def _remove_cmake_comments(content):
+    def replacer(match):
```

Comment:
Please reference the original code: https://stackoverflow.com/questions/241327/python-snippet-to-remove-c-and-c-comments#241506